client/my-sites/themes/theme-preview.jsx

```diff
@@ -2,79 +2,149 @@
  * External dependencies
  */
 import React from 'react';
+import { connect } from 'react-redux';
 import noop from 'lodash/noop';
 
 /**
  * Internal dependencies
  */
-import WebPreview from 'components/web-preview';
 import Button from 'components/button';
-import { getPreviewUrl } from './helpers';
+import QueryTheme from 'components/data/query-theme';
+import { connectOptions } from './theme-options';
+import { getThemeForPreviewData, getTheme, getThemePreviewState } from 'state/themes/selectors';
+import { getPreviewUrl } from 'my-sites/themes/helpers';
+import { localize } from 'i18n-calypso';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import { closeThemePreview } from 'state/themes/actions';
+import WebPreview from 'components/web-preview';
 
-export default React.createClass( {
+const ThemePreview = React.createClass( {
 	displayName: 'ThemePreview',
 
 	propTypes: {
 		theme: React.PropTypes.object,
 		showPreview: React.PropTypes.bool,
 		showExternal: React.PropTypes.bool,
 		onClose: React.PropTypes.func,
-		primaryButtonLabel: React.PropTypes.string.isRequired,
-		onPrimaryButtonClick: React.PropTypes.func,
 		getPrimaryButtonHref: React.PropTypes.func,
 		secondaryButtonLabel: React.PropTypes.string,
 		onSecondaryButtonClick: React.PropTypes.func,
-		getSecondaryButtonHref: React.PropTypes.func,
 	},
 
 	getDefaultProps() {
 		return {
-			onPrimaryButtonClick: noop,
 			getPrimaryButtonHref: () => null,
 			onSecondaryButtonClick: noop,
-			getSecondaryButtonHref: () => null,
 		};
 	},
 
 	onPrimaryButtonClick() {
-		this.props.onPrimaryButtonClick( this.props.theme );
-		this.props.onClose();
+		const option = this.getPrimaryOption();
+		option.action && option.action( this.props.theme );
+		this.props.closeThemePreview();
 	},
 
 	onSecondaryButtonClick() {
-		this.props.onSecondaryButtonClick( this.props.theme );
-		this.props.onClose();
+		const secondary = this.getSecondaryOption();
+		secondary.action && secondary.action( this.props.theme );
+		this.props.closeThemePreview();
+	},
+
+	getPrimaryOption() {
+		const { translate } = this.props;
+		const { purchase } = this.props.options;
+		const { price } = this.props.theme;
+		let { primary } = this.props.previewData.themeOptions;
+		primary.label = translate( 'Activate this design' );
+
+		if ( price && purchase ) {
+			primary = purchase;
+			primary.label = translate( 'Purchase this design' );
```

Comment:
Hi! I've found a possible matching string that has already been translated **31** times:
`translate( 'Purchase this theme' )` *ES Score: 6.37*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).